dist/src/main/java/io/camunda/commons/job/JobHandlerConfiguration.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.commons.job;
+
+import io.camunda.zeebe.broker.client.api.BrokerClient;
+import io.camunda.zeebe.gateway.impl.configuration.LongPollingCfg;
+import io.camunda.zeebe.gateway.impl.job.ActivateJobsHandler;
+import io.camunda.zeebe.gateway.impl.job.LongPollingActivateJobsHandler;
+import io.camunda.zeebe.gateway.impl.job.RoundRobinActivateJobsHandler;
+import io.camunda.zeebe.gateway.protocol.rest.JobActivationResponse;
+import io.camunda.zeebe.gateway.rest.ResponseMapper;
+import io.camunda.zeebe.gateway.rest.controller.JobActivationRequestResponseObserver;
+import io.camunda.zeebe.gateway.rest.controller.ResponseObserverProvider;
+import io.camunda.zeebe.scheduler.Actor;
+import io.camunda.zeebe.scheduler.ActorScheduler;
+import java.util.concurrent.CompletableFuture;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.util.unit.DataSize;
+
+@Configuration(proxyBeanMethods = false)
+@ConditionalOnProperty(
+    prefix = "zeebe.broker.gateway",
+    name = "enable",
+    havingValue = "true",
+    matchIfMissing = true)
```

Comment:
Using `@ConditionalOnRestGatewayEnabled` cannot be used, while the conditional annotation matches if the bean `RestGatewayDisabled` is missing, it does not declare in which order the beans are registered and initiated by Spring. Meaning it happens that this `JobHandlerConfiguration` is configured first before actually the `RestGatewayDisabled` is initiated. In that case, it would start the job activation handler which would not be expected.